### ISSUE TITLE
Bug 369296 - [nls tooling] Externalize Strings cannot use src/main/resources 

### ISSUE THREAD

**Issue by hd42** [eclipse-m2e/m2e-core#139]
See also: https://bugs.eclipse.org/bugs/show_bug.cgi?id=369296

Problem description from Stephen Cooper from the original ticket is still relevant in Eclipse 21.3:

'I have a maven project which uses standard directory structures of src/main/java and src/main/resources.

I want my messages.properties file to be generated in src/main/resources instead of the default src/main/java.

When I press "Configure..." and try to change the source folder for the property file location to src/main/resources, I get the message "The specified package for the resource bundle is invalid".

Yes, there are work-arounds which involve moving the messages.properties file back and forth between the src/main/java and src/main/resources directories, but I feel that this is something Eclipse should support out of the box.

I'm using the m2e plugin to manage my project configuration.

Reproducible: Always

Steps to Reproduce:
1. Select a project which is managed by maven/m2e plugin.
2. Select Source->Externalize Strings...
3. Select a class with strings to externalize
4. press Externalize...
5. press Configure...
6. Change the source folder for the property file location to src/main/resources
7. Get the message "The specified package for the resource bundle is invalid".'

If I understand the comments in the original issue correctly, the problem is that src/main/resources is "not a true Java source folder" and therefore the Externalize Strings tool can't handle it.

A fix would probably required coordinated changes in JDT UI (to allow certain non-java-source-folders as target for properties files) and m2e (to tell JDT about src/main/resources as a folder to be used).

**Comment by laeubi** [comment]
@hd42 just wondering _src/main/resources_ should be a source folder anyways. is it marked as such? does it work to manually mark it as a source folder?

**Comment by hd42** [comment]
@laeubi, yes, it is a source folder, but m2e sets an "**" exclusion pattern, which seems to lead to the Externalize Strings wizard to not recognize it as a 'true' source folder. I did not debug the Wizard code, only reproduced the behavior documented in https://bugs.eclipse.org/bugs/show_bug.cgi?id=369296

**Comment by laeubi** [comment]
Does it work if you remove the exclusion? The it might be more an JDT issue than an m2e issue. I think this exclusion is to prevent JDT to try compile anything in the folder, but this should not prevent the Wizard from placing files into it.  My guess is that it tries to not only place the properties but also the messages class there.

**Comment by hd42** [comment]
Yes, removing the exclusion works. The Messages-class is generated in src/main/java and the messages.properties in src/main/resources.
So, should m2e maybe not generate that exclusion?
